onnxruntime/core/providers/cpu/tensor/tile.cc

```diff
@@ -24,20 +24,49 @@ namespace onnxruntime {
 ONNX_CPU_OPERATOR_KERNEL(
     Tile,
     6,
-    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
-    Tile<float>);
+    KernelDefBuilder().TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
+                                            DataTypeImpl::GetTensorType<int32_t>()}),
+    Tile);
 
-template <>
-Status Tile<float>::Compute(OpKernelContext* ctx) const {
+
+template <typename T>
```

Comment:
We need to get rid of templates due to binary size growing. Just rewrite this generically with element_size as a differential. The only copy of the code should be for std::string bc it is special other than that it should all be non-templated code for all types (except string). See Transpose as an inspiration.

Reply:
I had your transpose change in mind but I thought I would do it it another change so as to get people unblocked. I de-templatized in this PR. I focussed on the fixed type tensors citing the urgency. I will support string/ float16 in an upcoming change.